### PR TITLE
Schedule Recurring Pickup + Summary UI

### PR DESCRIPTION
**Pull request by huangjoyce3**
image
image

### CODE REVIEW

File: src/Map/Map.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import React, { Component } from 'react';
+import GoogleMap from 'google-map-react';
+
+class Map extends Component{
+    constructor(props){
+        super(props);
+        this.state  = {
```
Comment (arielin3) [inline src/Map/Map.js]: just curious, what does this default to?

File: src/PageContent/RequestPickup/RecurringPickupRequest.js [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    }
+
+    componentDidMount(){
+        const daRef = firebase.database().ref('donating_agencies').child(this.props.account.agency);
+        daRef.on('value', (snapshot) => {
+            let donatingAgency = snapshot.val();
+            this.setState({
+                schoolID: donatingAgency.school
+            });
+        });
```
Comment (arielin3) [inline src/PageContent/RequestPickup/RecurringPickupRequest.js]: you're not using this right? if not, it'd be best if you could remove it so it's clearer for me when I add backend code

File: src/PageContent/RequestPickup/RecurringPickupRequest.js [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        let errors = {};
+        let formIsValid = true;
+
+        //Date
+        if(!fields['startDate']){
+            formIsValid = false;
+            errors['startDate'] = 'Start date cannot be empty';
+        }
+
+        if(fields['endDate'] < fields['startDate']){
```
Comment (arielin3) [inline src/PageContent/RequestPickup/RecurringPickupRequest.js]: are the values of type String or Date? You can compare like this? (If yes, that's cool!)

File: src/PageContent/RequestPickup/RecurringPickupRequest.js [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+            formIsValid = false;
+            errors['startTime'] = 'Start time cannot be empty';
+        }
+
+        if(!fields['endTime']){
+            formIsValid = false;
+            errors['endTime'] = 'End time cannot be empty';
+        }
+
+        if(fields['startTime'] > fields['endTime'] || fields['endTime'] < fields['startTime']){
```
Comment (arielin3) [inline src/PageContent/RequestPickup/RecurringPickupRequest.js]: aren't the two checks the same thing?

File: src/PageContent/RequestPickup/RecurringPickupRequest.js [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+        return (
+            <div className="form" id="recurringForm">
+                <form onSubmit={this.createRequest}>
+                    <div className="info">
+                        <p id="form-heading">Schedule Recurring Pickup</p>
+                        <p className="error">{this.state.errors['endBeforeStart']}</p>
+                        <p className="error">{this.state.errors['startTime']}</p>
+                        <p className="error">{this.state.errors['endTime']}</p>
+                        <p className="error">{this.state.errors['startDate']}</p>
+                        <p className="error">{this.state.errors['time']}</p>
```
Comment (arielin3) [inline src/PageContent/RequestPickup/RecurringPickupRequest.js]: so if there aren't any values in `this.state.errors`, these will be empty `<p></p>`s?

File: src/PageContent/RequestPickup/RecurringPickupRequest.js [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+        let fields = this.state.fields;
+        let errors = {};
+        let formIsValid = true;
+
+        //Date
+        if(!fields['startDate']){
+            formIsValid = false;
+            errors['startDate'] = 'Start date cannot be empty';
+        }
+
```
Comment (arielin3) [inline src/PageContent/RequestPickup/RecurringPickupRequest.js]: `endDate` actually isn't required, it depends on which `durationType` the user chooses:

1. `durationType`, the radio button for "Recur" or "End by" is required
2. If chose "Recur", the `numRecurrences` input box can't be empty
3. If chose "End By", the `endDate` input can't be empty.

So you'll want to:

- check the radio button isn't empty
- check the corresponding input isn't empty
- only check `endDate` against `startDate` when "End By" is chosen
- (could add to TODO for this component) check for both cases that it's within 1 year.

File: src/PageContent/RequestPickup/RecurringPickupRequest.js [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+    // write to firebase
+    // var newRequest = firebase.database().ref().child("delivery_requests").push();
+    // newRequest.set(deliveryRequest);
+    
+    submitRequest(){
+        
+    }
+
+    createRequest(event) {
```
Comment (arielin3) [inline src/PageContent/RequestPickup/RecurringPickupRequest.js]: what about the `Cancel` case?

File: src/PageContent/RequestPickup/RecurringPickupRequest.js [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+
+    handleChange(field, e){         
+        let fields = this.state.fields;
+        fields[field] = e.target.value;        
+        this.setState({fields});
+    }
+
+    toggleModal(){
+        this.setState({
+            isOpen: !this.state.isOpen
```
Comment (arielin3) [inline src/PageContent/RequestPickup/RecurringPickupRequest.js]: same as in Sarah's PR, can't depend on values in `this.state` in `setState()`. See  https://reactjs.org/docs/faq-state.html#why-is-setstate-giving-me-the-wrong-value

File: src/PageContent/RequestPickup/RecurringPickupRequest.js [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+                <PickupSummary 
+                    type={'Request Recurring Pickup'} 
+                    startDate={this.state.formInfo.startDate} 
+                    endDate={this.state.formInfo.duration}
+                    startTime={this.state.formInfo.startTime}
+                    endTime={this.state.formInfo.endTime}
+                    notes={this.state.formInfo.notes}
+                    account={this.props.account}
+                    show={this.state.isOpen} 
+                    onClose={this.toggleModal.bind(this)}
```
Comment (arielin3) [inline src/PageContent/RequestPickup/RecurringPickupRequest.js]: bind it in constructor

File: src/PageContent/RequestPickup/PickupSummary.js [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+class PickupSummary extends React.Component {
+    render() {
+        // Render nothing if the "show" prop is false
+        if(!this.props.show) {
+            return null;
+        }
+        return (
+            <div className="backdrop">
+                {/* TODO: fix background opacity. Maybe with iFrame. */}
```
Comment (arielin3) [inline src/PageContent/RequestPickup/PickupSummary.js]: remember to add this to the TODOs for this component

File: src/PageContent/RequestPickup/PickupSummary.js [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+                    <p id="exit" onClick={this.props.onClose}>&times;</p>
+                    <div className="summary-title flex">
+                        <img src={truck} alt="truck"></img>
+                        <p id="title">{this.props.type}</p>
+                    </div>
+                    <div className="wrapper grid">
+                        <div className="details grid">
+                            <p id="subheading">Pickup Details</p>
+                            <p>Start Date: {this.props.startDate}</p>
+                            {/* TODO: determine end date if they chose recurring # times option */}
```
Comment (arielin3) [inline src/PageContent/RequestPickup/PickupSummary.js]: actually, we don't always show end date. It depends on what they chose:

- if chose "End by", display something like: "every other Thursday until 10/23/2018"
- if chose "Recur", display "10 pickups requested for every other Thursday" (see Figma)

where,
- "every other" depends on the `repeats` they chose
- "Thursday" depends on the week day of the `startDate`

File: src/PageContent/RequestPickup/PickupSummary.js [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+                            <div className="agency grid">
+                                <p id="subheading">Dining Hall</p>
+                                <p id="name">Local Point</p>
+                                <div className="contact">
+                                    <p>Andrea</p>
+                                    <p>###-###-####</p>
+                                    <p>email</p>
+                                </div>
+                            </div>
+                            <Map></Map>
```
Comment (arielin3) [inline src/PageContent/RequestPickup/PickupSummary.js]: add TODO to populate map correctly?

File: src/PageContent/RequestPickup/PickupSummary.js [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+                        </div>
+                        <div className="details grid">
+                            <p id="subheading">Notes for Pickup</p>
+                            <p>{this.props.notes}</p>
```
Comment (arielin3) [inline src/PageContent/RequestPickup/PickupSummary.js]: i don't think `notes` is required, so should probably only display if there are notes to display. (could add to TODO if you want)

File: src/PageContent/RequestPickup/RecurringPickupRequest.js (hunk 5 above)
Comment (huangjoyce3) [reply src/PageContent/RequestPickup/RecurringPickupRequest.js]: yes

File: src/PageContent/RequestPickup/RecurringPickupRequest.js (hunk 3 above)
Comment (huangjoyce3) [reply src/PageContent/RequestPickup/RecurringPickupRequest.js]: They are type Date. I set fields in my handleChange function 
```
handleChange(field, e){         
        let fields = this.state.fields;
        fields[field] = e.target.value;        
        this.setState({fields});
}
```

File: src/PageContent/RequestPickup/RecurringPickupRequest.js (hunk 8 above)
Comment (huangjoyce3) [reply src/PageContent/RequestPickup/RecurringPickupRequest.js]: then how to go about it?

**Pull request closed without merging**